File: webflux-dialogue/src/main/java/com/study/webflux/rag/domain/model/llm/TokenUsage.java
package com.study.webflux.rag.domain.model.llm;

public record TokenUsage(
	int promptTokens,
	int completionTokens,
	int totalTokens
) {
	public static TokenUsage zero() {
		return new TokenUsage(0, 0, 0);
	}

	public static TokenUsage of(int promptTokens, int completionTokens) {
		return new TokenUsage(promptTokens, completionTokens, promptTokens + completionTokens);
	}
}

File: app-miyou/src/main/kotlin/com/miyou/app/domain/dialogue/model/TokenUsage.kt
package com.miyou.app.domain.dialogue.model

data class TokenUsage(
    val promptTokens: Int,
    val completionTokens: Int,
    val totalTokens: Int,
) {
    companion object {
        fun zero(): TokenUsage = TokenUsage(0, 0, 0)

        fun of(
            promptTokens: Int,
            completionTokens: Int,
        ): TokenUsage = TokenUsage(promptTokens, completionTokens, promptTokens + completionTokens)
    }
}
